dockstore-webservice/src/main/resources/migrations.1.16.0.xml

```diff
@@ -316,4 +316,19 @@
             WHERE version_metadata.doiurl IS NOT NULL;
         </sql>
     </changeSet>
+    <changeSet author="svonworl" id="addFormFieldToSourceFile">
+        <addColumn tableName="sourcefile">
+            <column name="form" type="VARCHAR(255)" value="COMPLETE">
+                <constraints nullable="false"/>
+            </column>
+        </addColumn>
+        <sql dbms="postgresql">
+            <comment>Set appropriate content and form for sourcefiles with null content</comment>
+            UPDATE sourcefile SET form = 'ERROR', content = 'Dockstore could not retrieve this file' WHERE content IS NULL;
```

Comment:
Do we know that this is always the reason why the content is null? Just checking... And there are no null content source files for hosted entries?

Reply:
Oh, upon further investigation, turns out that we do intentionally create SourceFile stubs, in `EntryVersionHelper` at least.  I've added a `STUB` state and changed the migrations to define all existing null-content SourceFiles as stubs.  The webservice might not have intentionally create them all as stubs, but that's effectively what they are now.

Reply:
Editorial: the "stub" concept introduces another state into our system, which makes it more difficult to reason about.  If we can simplify our system by someday eliminating stubs, that probably would be good,